plugins/hr-resources/src/components/ScheduleView.svelte

```diff
@@ -230,32 +232,82 @@
       'date.year': currentDate.getFullYear()
     },
     (res) => {
-      holidays = res.map((holiday) => new Date(fromTzDate(holiday.date)))
+      const group = groupBy(res, 'department')
+      holidays = new Map()
+      for (const groupKey in group) {
+        holidays.set(
+          groupKey,
+          group[groupKey].map((holiday) => new Date(fromTzDate(holiday.date)))
+        )
+      }
     }
   )
+
+  async function getHolidays (month: Date): Promise<Map<Ref<Department>, Date[]>> {
+    const result = await client.findAll(hr.class.PublicHoliday, {
+      'date.month': month.getMonth(),
+      'date.year': month.getFullYear()
+    })
+    const group = groupBy(result, 'department')
+    const rMap = new Map()
+    for (const groupKey in group) {
+      rMap.set(
+        groupKey,
+        group[groupKey].map((holiday) => new Date(fromTzDate(holiday.date)))
+      )
+    }
+    return rMap
+  }
+
+  let staffDepartmentMap: Map<Ref<Staff>, Department[]> = new Map<Ref<Staff>, Department[]>()
+  const client = getClient()
+
+  async function getDepartmentsForEmployee (): Promise<void> {
+    staffDepartmentMap = new Map<Ref<Staff>, Department[]>()
+    if (departmentStaff.length) {
+      const ids = departmentStaff.map((staff) => staff._id)
+      for (const id of ids) {
+        const staffs = await client.findOne(contact.class.EmployeeAccount, { employee: id })
```

Comment:
Better to have one findAll for all Employee and all depertments and do mapping instead of so many find calls.